pkg/controller/factory/factory.go

```diff
@@ -54,6 +77,26 @@ func (f *Factory) WithInformers(informers ...Informer) *Factory {
 	return f
 }
 
+// WithInformerQueueKeys allows to pass multiple InformerQueueKeyFunc that return informer and the corresponding queue key function
+// that will be used for the particular informer.
+func (f *Factory) WithInformerQueueKeyFuncs(mappings ...InformerQueueKeyFunc) *Factory {
+	f.informerQueueKeys = append(f.informerQueueKeys, mappings...)
+	return f
+}
+
+// WithQueueKeyFunc allows to specify custom queue key function.
+// By default, the queue key func returns "key" constant.
+func (f *Factory) WithStringQueueKeyFunc(queueKey StringQueueKeyFunc) *Factory {
```

Comment:
Why do we need a string transformer, i.e. no constant string key func?

Reply:
we don't need this and we don't use it anywhere... i think we don't need to allow controllers to customize the queue "key" value at all. i removed this